crates/sui-indexer-alt-framework-store-traits/src/lib.rs

```diff
@@ -81,7 +86,19 @@ pub trait Store: Send + Sync + 'static + Clone {
     where
         Self: 'c;
 
+    /// Delimiter used to separate pipeline names from task identifiers when reading or writing the
+    /// committer watermark.
+    const DELIMITER: &'static str = "@";
+
     async fn connect<'c>(&'c self) -> Result<Self::Connection<'c>, anyhow::Error>;
+
+    /// Constructs the watermark key used to read or write the committer watermark.
+    fn watermark_key(pipeline: &str, task: Option<&str>) -> String {
```

Comment:
nit: should this also be called?

```suggestion
    fn pipeline_task(pipeline: &str, task: Option<&str>) -> String {
```

Also, should this be part of the trait? It means that someone can override this behaviour (as well as set a custom `DELIMITER`), which could cause unexpected problems, because the code that checks pipeline name validity is hardcoded elsewhere to check for the non-existence of the `DELIMITER`.

It's also possible that we want to lean in more here by only offering `pipeline_task` as a way to customise how the name is generated, as well as a separate `is_valid_pipeline` function and default them to:

```rust
fn pipeline_task(pipeline: &str, task: Option<&str>) -> String {
    match task {
      Some(task) => format!("{pipeline}@{task}"),
      None => pipeline.to_owned()
    }
}

fn is_valid_pipeline(pipeline: &str) -> bool {
    !pipeline.contains('@')
}
```

I'm fine either way, but the hybrid doesn't really work.